sdk/nodejs/introspector/analysis/analysis.ts

```diff
@@ -0,0 +1,179 @@
+import ts from "typescript"
+
+import { UnknownDaggerError } from "../../common/errors/UnknownDaggerError.js"
+import {
+  ClassMetadata,
+  FunctionMetadata,
+  Metadata,
+  PropertyMetadata,
+} from "./metadata.js"
+import { serializeSignature, serializeSymbol } from "./serialize.js"
+import { isFunction, isObject, isPublicProperty } from "./utils.js"
+
+/**
+ * Analyse the list of Typescript File using the Typescript compiler API.
+ *
+ * This function introspect files and returns metadata of their class and
+ * functions that should be exposed to the Dagger API.
+ *
+ * WARNING(28/11/23): This does NOT include arrow style function.
+ *
+ * @param files List of Typescript files to introspect.
+ */
+export function analysis(files: string[]): Metadata {
```

Comment:
Why is it needed to load source files for introspection? Python relies on the SDK importing the user's module from a conventionalized location (currently `main`), which will execute the decorators just by importing. The decorators then add the decorated object to a registry which is available during execution time.

From the user's main file, other files can be imported there.

Reply:
Resolved in [27477bc](https://github.com/dagger/dagger/pull/6159/commits/27477bc051006373f25a0ad00364ef1e6bc95bea)